### DISCUSSION TITLE
Database: Error 1205: Lock wait timeout exceeded; try restarting transaction

### DISCUSSION THREAD

**Discussion by RobHofmann** [photoprism/photoprism#2171] [Q&A]
Hi,

The past few days i've been playing around with Photoprism a bit. I'm trying to load about 75k photos into Photoview. There are two things i'm noticing;
1. After scanning (even complete rescan or doing a clean install; tried multiple times) it only shows ~58k photos & videos. I have no clue what happened to the rest. Is there any place for logs for failed imports?
2. After doing `docker exec photoprism photoprism faces index`, i'm seeing errors after 3 hours. See these logs:
```
<loads of image processings>
time="2022-03-20T02:49:51Z" level=warning msg="index: Error 1205: Lock wait timeout exceeded; try restarting transaction (update counts)"
time="2022-03-20T02:51:05Z" level=error msg="index: Error 1205: Lock wait timeout exceeded; try restarting transaction (update primary files)"
time="2022-03-20T02:51:55Z" level=error msg="Error 1205: Lock wait timeout exceeded; try restarting transaction"
time="2022-03-20T02:51:55Z" level=info msg="indexed 86,135 files in 3h0m45.766240881s"
time="2022-03-20T02:51:55Z" level=info msg="closed database connection"
```
Also note that it seems to crash after 3 hours with a timeout. Is there a way to increase the timeout?
3. I also see a lot of `index: skipped main <filetype> file `. How can I force to take all files (also in a stack).
4. Scanning & facial recognition seems to take a long time. I reckon this would go faster with a NVIDIA GPU, right? But the real question is; does the facial recognition also get better (in terms of matches & quality) with a GPU?

So in short: I'm missing a lot of photos and also a lot of faces are not detected or not matched correctly. I've looked through the options, but can't find any options (other than the facial recognition tweaks) which can help me in the right direction.

Is there anything I can do about this? Looking forward to hear from you.

PS. I am really impressed so far with what i've seen!

**Comment by lastzero** [comment]
https://docs.photoprism.app/getting-started/troubleshooting/performance/#troubleshooting

**Comment by RobHofmann** [comment]
This has a lot of nice best practises (which i've applied), but as far as I understand, it does not answer my questions, right? Sorry if  I dont understand.

**Comment by graciousgrey** [comment]
Hi :)

1. Do you see files in the _Hidden_ or errors in the _Errors_ section ? Here is the complete list with possible reason why images could be missing: https://docs.photoprism.app/getting-started/troubleshooting/#missing-pictures. 
See also our FAQs on the file counts: https://docs.photoprism.app/user-guide/faq/#library-counts-files

2. Did you run faces index at the same time as the normal index? Note that the normal index also indexes faces unless you have turned it off in settings. Indexing faces multiple times won't lead to more faces being detected.

3. `index: skipped main <filetype> file` This is expected the normal index skips files that are already indexed and not modified

4. GPU will not improve matches. Facial recognition works best the more images you have. A high thumbnail quality leads to the best results, so we recommend using at least standard settings: https://docs.photoprism.app/user-guide/settings/advanced/#images

Facial detection and recognition is never perfect. PhotoPrism will update face clusters in the background when you start adding names to faces or rejecting wrong faces from a person.  Many improvements are planned as well:  https://github.com/photoprism/photoprism/issues?q=is%3Aissue+is%3Aopen+people+in%3Atitle

I hope this helps!

**Comment by RobHofmann** [comment]
Hi! Thanks for taking the time to help me out! This clears up some of my questions!

1. I do see 22 hidden files (they seem broken, but its fine. its just 22 files out of 72kish).
![image](https://user-images.githubusercontent.com/4020931/159376284-2d89e5a4-3499-4410-91fa-a795f78ce115.png)
I do see warnings in the error logs:
![image](https://user-images.githubusercontent.com/4020931/159376382-6d882552-e802-49f4-bc9d-0adf144441d8.png)
Also 2 errors on top (which match which i posted earlier):
![image](https://user-images.githubusercontent.com/4020931/159376422-a65a7836-456f-4ca9-9ed9-51d64eb305bb.png)
2. No I did a complete rescan. Let the indexing finish and the next day I did another index in the hope it would find more; it didnt.
3. Alright this is clear.
4. Alright. So it simply makes it faster. The issue at this point is that it takes a vast amount of space to store the created thumbnails. I fear if i create better/bigger thumbnails this will cause issues. I'm at 81GB right now. Would it (maybe in theorie, because this feature does not exist yet) be possible to take the source image to scan it for faces rather than the generated thumbnail? Or am I saying something which does not make sense. I wouldnt care for the processing to take weeks. Also I could add a GPU to improve the speed.

Another thing i'm noticing since today is that most of the sections open pretty fast (`Search` or `Videos` or `Moments` or `Places`). Whenever I try to open a recognized person, it will keep loading forever (it seems to be stuck at MySQL). This is what `show full processlist;` on MySQL shows me:
![image](https://user-images.githubusercontent.com/4020931/159377089-49ec9ae8-ffd3-4b81-ace1-a7b09f73ee91.png)
This worked before 🤔. I've been tweaking some settings on MySQL (increasing memory based on the mysql tuner script etc).

**Reply by RobHofmann** [reply]
Small addition; i'm not sure if this might be the issue, but i'm mounting the photos folder as `:ro`. Does this mean it can't create JPEGs from (for example) PNG's which will cause it not to show the photo at all?

If so: is there a way to create the JPEGs in another folder than my source folder (I rather keep the source folder untouched).

**Reply by srett** [reply]
JPEGs (Thumbnails, Video Thumbnails, RAW Conversions etc.) are created in the storage folder. PP doesn't need write access to your originals folder unless you run it in RW mode (which is for being able to delete and import photos), or if you set the sidecar path to e.g. "."

**Comment by pabsi** [comment]
I was able to stop this from happening by setting this on my MariaDB instance:
`set global innodb_lock_wait_timeout=300` (by default it's 50).

Or edit the config file `my.cnf` (in my case `/etc/mysql/mariadb.conf.d/50-server.cnf`)

This was my Errors page before:
![image](https://user-images.githubusercontent.com/5247611/170317191-0b26135c-8e43-426b-ae11-94902af2e0cf.png)
